lib/clientsidescripts.js

```diff
@@ -44,33 +44,37 @@ function wrapWithHelpers(fun) {
  * Asynchronous.
  *
  * @param {string} rootSelector The selector housing an ng-app
- * @param {boolean} ng12Hybrid Flag set if app is a hybrid of angular 1 and 2
  * @param {function(string)} callback callback. If a failure occurs, it will
  *     be passed as a parameter.
  */
-functions.waitForAngular = function(rootSelector, ng12Hybrid, callback) {
+functions.waitForAngular = function(rootSelector, callback) {
   var el = document.querySelector(rootSelector);
 
   try {
-    if (!ng12Hybrid && window.getAngularTestability) {
+    if (window.angular && !(window.angular.version &&
+        window.angular.version.major > 1)) {
+      if (angular.getTestability) {
+        angular.getTestability(el).whenStable(callback);
+      } else if (angular.element(el).injector()) {
+        angular.element(el).injector().get('$browser').
+            notifyWhenNoOutstandingRequests(callback);
+      } else {
+        throw new Error('root element (' + rootSelector + ') has no injector.' +
+           ' this may mean it is not inside ng-app.');
+      }
+    } else if (window.getAngularTestability) {
       window.getAngularTestability(el).whenStable(callback);
-      return;
-    }
-    if (!window.angular) {
+    } else if (!window.angular) {
       throw new Error('window.angular is undefined.  This could be either ' +
           'because this is a non-angular page or because your test involves ' +
           'client-side navigation, which can interfere with Protractor\'s ' +
           'bootstrapping.  See http://git.io/v4gXM for details');
-    }
-    if (angular.getTestability) {
-      angular.getTestability(el).whenStable(callback);
+    } else if (window.angular.version == 2) {
```

Comment:
Since Angular versions are going to keep advancing now, we should change this to:

```js
else if (window.angular.version.major >= 2) {
  throw new Error('You appear to be using angular, but window.getAngularTestability was ' + 
     'never set. This may be due to bad obfuscation.')
```